llvm/lib/Target/SPIRV/SPIRVPreTranslationLegalizer.cpp

```diff
@@ -262,8 +268,8 @@ bool SPIRVPreTranslationLegalizer::runOnFunction(Function *Func,
 
   for (auto *I : Worklist) {
     bool TrackConstants = true;
-    if (!I->getType()->isVoidTy())
-      B.SetInsertPoint(I->getNextNode());
+    if (!I->getType()->isVoidTy() || isa<StoreInst>(I))
+      setInsertPointAfterPhi(B, I->getNextNode());
```

Comment:
same as above

Reply:
This (B.SetInsertPoint(I->getParent()->getFirstNonPHI());) will insert always at the beginning of BB after all phis. So the substitution does not work in the case of non-phi instructions in the middle of BB, since it can make a reg use before it's defined.

Reply:
Yep, that just means we don't need setInsertPointAfterPhi everywhere since PHIs should be grouped at the top of the BB

Reply:
Well, I checked that only two of setInsertPointSkippingPhis are really requied, and removed others.